Assets/Scripts/EntityBehaviour.cs

```diff
@@ -0,0 +1,46 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EntityBehaviour : MonoBehaviour
+{
+    [SerializeField] private int enemyType;
+
+    [SerializeField] private int health;
+    
+
+    public void Trigger_TakeDamage(int _bulletType)
+    {
+        //TAKE DAMAGE
+        if (_bulletType == enemyType)
+            Trigger_InstantKill();
+
+        else health--;
+
+        //CHECK FOR HEALTH/DEATH
+        if (health <= 0)
+            Death();
+
+        else
+        {
+            //TAKING HIT ANIMATION AND SOUND CAN BE INSERTED HERE
```

Comment:
this is only justified if you think you will never find this again. the function name is good enough that we don't have to be pointed specifically here, we're all programmers after all

Reply:
haha this is a reminder to myself!